### PR TITLE
AF-923: cache invalidation on session expire/logout for security related info

### PR DESCRIPTION
**Pull request by porcelli**
@domhanak @ederign bringing feature to master 

looks like due the changes that @dgutierr was executing on security, this feature was postponed and never revisited.

### CODE REVIEW

File: uberfire-extensions/uberfire-security/uberfire-servlet-security/src/main/java/org/uberfire/ext/security/server/CleanupSecurityCacheSessionListener.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    public void sessionCreated(HttpSessionEvent se) {
+    }
+
+    @Override
+    public void sessionDestroyed(HttpSessionEvent se) {
+        if (authorizationManagers.isEmpty()) {
+            return;
+        }
+        final User currentUser = (User) se.getSession().getAttribute(ServletSecurityAuthenticationService.USER_SESSION_ATTR_NAME);
+        if (currentUser != null && !currentUser.equals(User.ANONYMOUS)) {
```
Comment (jomarko) [inline uberfire-extensions/uberfire-security/uberfire-servlet-security/src/main/java/org/uberfire/ext/security/server/CleanupSecurityCacheSessionListener.java]: Can be simplified as `!Objects.equals(currentUser, User.ANONYMOUS)`

**Comment by domhanak** [comment]
jenkins execute full downstream build

File: uberfire-extensions/uberfire-security/uberfire-servlet-security/src/main/java/org/uberfire/ext/security/server/CleanupSecurityCacheSessionListener.java (hunk 1 above)
Comment (porcelli) [reply uberfire-extensions/uberfire-security/uberfire-servlet-security/src/main/java/org/uberfire/ext/security/server/CleanupSecurityCacheSessionListener.java]: thank you for spot it, I did simplify it by using equals of the User.ANONYMOUS object

Comment (jomarko) [reply uberfire-extensions/uberfire-security/uberfire-servlet-security/src/main/java/org/uberfire/ext/security/server/CleanupSecurityCacheSessionListener.java]: ok, thank you for update

**Comment by domhanak** [comment]
jenkins execute full downstream build

**Comment by domhanak** [comment]
@porcelli I will try it and verify it with upstream today if the build comes through

**Comment by domhanak** [comment]
jenkins execute full downstream build

**Comment by ederign** [comment]
@domhanak did you try it?

**Pull request merged**